server/src/main/java/org/opensearch/indices/store/ShardAttributes.java

```diff
@@ -12,36 +12,27 @@
 import org.opensearch.core.common.io.stream.StreamInput;
 import org.opensearch.core.common.io.stream.StreamOutput;
 import org.opensearch.core.common.io.stream.Writeable;
-import org.opensearch.core.index.shard.ShardId;
-import org.opensearch.gateway.AsyncShardFetch;
 
 import java.io.IOException;
 
 /**
- * This class contains information about the shard that needs to be sent as part of request in Transport Action implementing
- * {@link AsyncShardFetch.Lister} to fetch shard information in async manner
+ * This class contains Attributes related to Shards that are necessary for making the
+ * {@link org.opensearch.gateway.TransportNodesListGatewayStartedShards} transport requests
  *
  * @opensearch.internal
  */
 public class ShardAttributes implements Writeable {
-    private final ShardId shardId;
```

Comment:
Removing shardId, as there is no need of it. ShardAttributes is already part of a map <ShardId, ShardAttributes> so this was just duplicate information.

Reply:
@shwetathareja  - class is being used for calling the list interface only, not in transport communication. It was already pushed long back for single shard flow already. But **shardId** parameter itself is not being used even before sending transport call, so removing it doesn't impact anything.

shardId parameter was only being used in batch transport, but that's not really needed if we do entry set traversal, as map's entry already has shardId.